### PR TITLE
feat(store): support dynamic plugins

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/ngxs/store/issues/835

## What is the new behavior?

Support dynamic state handlers

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by arturovt** [comment]
@splincode 
I like it! :tada:

File: docs/plugins/intro.md [hunk 1]
```diff
@@ -73,3 +73,24 @@ export class MyModule {}
 ```
 
 The method also works with `forFeature`.
+
+### Dynamic plugins
+
+You can also add your own plugins (state handlers) in runtime:
```
Comment (arturovt) [inline docs/plugins/intro.md]: at runtime

**Comment by splincode** [comment]
@markwhitfeld 

> Is there a particular use case that this is going to solve right now that cannot be solved with the existing plugins?

- Using dynamic import to reduce the size of the bundle

```ts
@Injectable()
class MyDynamicPluginManager {
  constructor(private pluginManager: PluginManager) {}

  public init(): void {
    if (!environment.production) {
      import('MY_PLUGIN_PATH//OR_LOGGER_PLUGIN//ETC').then((plugin) => {
         this.pluginManager.registerPlugin(plugin);
      });
    }
   }
}
```

**Comment by arturovt** [comment]
@splincode 

Using dynamic import can be used in cases when a plugin doesn't have any dependencies. But most plugins have.

**Pull request closed without merging**